### ISSUE TITLE
Runner label checking with expressions is fragile

### ISSUE THREAD

**Issue by rtarpine-vanta** [rhysd/actionlint#656]
The docs for runner label checking say that `Even if an expression is used in the section like runs-on: ${{ matrix.foo }}, actionlint parses the expression and resolves the possible values, then validates the values.`

This appears to only work in simple cases. For example, take the example from the playground:
```
    strategy:
      matrix:
        os: [macos-latest, linux-latest]
    runs-on: ${{ matrix.os }}
```
In the above case, actionlint correctly reports that `label "linux-latest" is unknown`. But if you use an expression in the `matrix`:
```
    strategy:
      matrix:
        os: [macos-latest, "${{'linux'}}-latest"]
    runs-on: ${{ matrix.os }}
```
then the problem is missed. Or if you use a more general expression in the `runs-on` itself:
```
    strategy:
      matrix:
        os: [macos, linux]
    runs-on: ${{ matrix.os }}-latest
```
it's also not reported.
